crates/store/re_grpc_client/src/address.rs

```diff
@@ -1,65 +1,111 @@
+use url::Url;
+
 /// The given url is not a valid Rerun storage node URL.
 #[derive(thiserror::Error, Debug)]
-#[error("URL {url:?} should follow rerun://addr:port/recording/12345")]
-pub struct InvalidAddressError {
-    url: String,
+#[error("URL {url:?} should follow rerun://addr:port/recording/12345 for recording or rerun://addr:port/catalog for catalog")]
+pub struct InvalidRedapAddress {
+    url: Url,
     msg: String,
 }
 
 /// Parsed `rerun://addr:port/recording/12345`
-pub struct Address {
-    pub addr_port: String,
+pub struct RecordingAddress {
+    pub redap_endpoint: Url,
     pub recording_id: String,
 }
 
-impl std::fmt::Display for Address {
+impl std::fmt::Display for RecordingAddress {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         write!(
             f,
-            "rerun://{}/recording/{}",
-            self.addr_port, self.recording_id
+            "RecordingAddress {{ redap_endpoint: {}, recording_id: {} }}",
```

Comment:
It was nice that previously display let us round-trip from str -> Address -> str, which meant we could print out addresses that could be copied back to the rerun-CLI input.

Reply:
yeah, agree. I wanted to verify early i.e. get `Url` in here, rather than just a string. I'll figure it out.